crates/pallet-executor/src/lib.rs

```diff
@@ -407,7 +409,9 @@ mod pallet {
                     let builder = ValidTransaction::with_tag_prefix("SubspaceSubmitBundle")
                         .priority(TransactionPriority::MAX)
                         .and_provides(first_primary_number)
-                        .longevity(CONFIRMATION_DEPTH_K as TransactionLongevity)
+                        .longevity(T::ConfirmationDepthK::get().try_into().unwrap_or_else(|_| {
+                            panic!("Block number always fits in TransactionLongevity; qed")
+                        }))
```

Comment:
Just curious, why did this have to change at all?

Reply:
The receipts in a bundle will become expired once the primary chain proceeds to a much higher block, hence it's reasonable to have a finite value for this tx so that the state receipts can be removed automatically.